homeassistant/components/tuya/climate.py

```diff
@@ -225,6 +225,16 @@ def __init__(  # noqa: C901
                 if tuya_mode in data_type.range:
                     self._hvac_to_tuya[ha_mode] = tuya_mode
                     self._attr_hvac_modes.append(ha_mode)
+
+            # Log unknown modes
+            for tuya_mode in data_type.range:
+                if tuya_mode in TUYA_HVAC_TO_HA.values():
+                    LOGGER.warning(
```

Comment:
We try to avoid side effects in init methods too. Is it possible to move it outside or maybe into `async_added_to_hass`?

Reply:
It is not dependent on `hass` in any way, this can be instanced incorrectly, regardless of `hass`; so `async_added_to_hass` isn't right either IMHO.

Reply:
I don't think there is an alternative place, considering the above, do you still think `async_added_to_hass` is the right place?

Reply:
I think it's better to not have side effects in init method, yes.

Reply:
Moved.